bigframes/core/nodes.py

```diff
@@ -574,6 +583,31 @@ def with_id(self, id: identifiers.ColumnId) -> ScanItem:
 class ScanList:
     items: typing.Tuple[ScanItem, ...]
 
+    def filter(
+        self,
+        ids: AbstractSet[identifiers.ColumnId],
+    ) -> ScanList:
+        result = ScanList(tuple(item for item in self.items if item.id in ids))
+        if len(result.items) == 0:
+            # We need to select something, or stuff breaks
+            result = ScanList(self.items[:1])
+        return result
+
+    def select(
```

Comment:
A docstring would be great. Are we talking about https://en.wikipedia.org/wiki/Selection_(relational_algebra) (aka row predicates) here?